crates/optimism/rpc/src/block.rs

```diff
@@ -0,0 +1,74 @@
+//! Loads and formats OP block RPC response.   
+
+use reth_primitives::TransactionMeta;
+use reth_provider::{BlockIdReader, BlockReader, ChainSpecProvider, ReceiptProvider};
+use reth_rpc::eth::{
+    api::{
+        block::EthBlocks,
+        transactions::{BuildReceipt, ReceiptBuilder},
+    },
+    error::EthResult,
+};
+use reth_rpc_types::{AnyTransactionReceipt, BlockId};
+
+use crate::receipt::op_receipt_fields;
+
+use super::OptimismApi;
+
+impl<Provider, Pool, Network, EvmConfig> EthBlocks
+    for OptimismApi<Provider, Pool, Network, EvmConfig>
+where
+    Provider: BlockIdReader + BlockReader + ReceiptProvider + ChainSpecProvider,
+{
+    #[inline]
+    fn provider(&self) -> &(impl BlockIdReader + BlockReader) {
```

Comment:
this can be combined into one, I think we have a super trait for these, I think `BlockReaderIdExt`